crowbar_framework/app/models/service_object.rb

```diff
@@ -1187,6 +1187,9 @@ def apply_role(role, inst, in_queue)
       return [405, message]
     end
 
+    # Invalidate cache as apply_role_pre_chef_call can save nodes
+    pre_cached_nodes = {}
```

Comment:
Actually, yes and no: we should change the line you mention, but not in that way: the cache can still be re-filled and that line should participate in that effort.